oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -387,17 +386,22 @@ public static ExternalAccountCredentials fromStream(InputStream credentialsStrea
    */
   public static ExternalAccountCredentials fromStream(
       InputStream credentialsStream, HttpTransportFactory transportFactory) throws IOException {
-    checkNotNull(credentialsStream);
-    checkNotNull(transportFactory);
-
-    JsonObjectParser parser = new JsonObjectParser(OAuth2Utils.JSON_FACTORY);
-    GenericJson fileContents =
-        parser.parseAndClose(credentialsStream, StandardCharsets.UTF_8, GenericJson.class);
-    try {
-      return fromJson(fileContents, transportFactory);
-    } catch (ClassCastException | IllegalArgumentException e) {
-      throw new CredentialFormatException("An invalid input stream was provided.", e);
+    Preconditions.checkNotNull(transportFactory);
+    GenericJson fileContents = parseJsonInputStream(credentialsStream);
+    String fileType = extractFromJson(fileContents, "type");
+    if (fileType.equals(GoogleCredentialsInfo.EXTERNAL_ACCOUNT_CREDENTIALS.getFileType())) {
+      try {
+        return fromJson(fileContents, transportFactory);
```

Comment:
@zhumin8 Talked with Sai and `ExternalAccountCredential.fromStream()` branching off to creating the subtypes is fine to keep.